gcloud-java-compute/src/main/java/com/google/gcloud/compute/DiskTypeId.java

```
@@ -0,0 +1,110 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.compute;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.MoreObjects;
+
+import java.util.Objects;
+
+/**
+ * Identity for a Google Compute Engine disk type.
+ */
+public final class DiskTypeId extends ZoneResourceId {
+
+  private static final long serialVersionUID = 7337881474103686219L;
+
+  private final String diskType;
+
+  DiskTypeId(String project, String zone, String diskType) {
+    super(project, zone);
+    this.diskType = checkNotNull(diskType);
+  }
+
+  /**
+   * Returns the name of the disk type resource. The name must be 1-63 characters long, and comply
+   * with RFC1035. Specifically, the name must be 1-63 characters long and match the regular
+   * expression {@code [a-z]([-a-z0-9]*[a-z0-9])?} which means the first character must be a
+   * lowercase letter, and all following characters must be a dash, lowercase letter, or digit,
+   * except the last character, which cannot be a dash.
+   *
+   * @see <a href="https://www.ietf.org/rfc/rfc1035.txt">RFC1035</a>
+   */
+  public String diskType() {
+    return diskType;
+  }
+
+  @Override
+  public String toUrl() {
+    return super.toUrl() + "/diskTypes/" + diskType;
+  }
+
+  @Override
+  MoreObjects.ToStringHelper toStringHelper() {
+    return super.toStringHelper().add("diskType", diskType);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(super.hashCode(), diskType);
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    return obj instanceof DiskTypeId && baseEquals((DiskTypeId) obj);
+  }
+
+  @Override
+  DiskTypeId setProjectId(String projectId) {
+    if (project() != null) {
+      return this;
```

I assume this means the DiskTypeId's project ID can't be changed.  Should this throw an exception instead of returning itself?